ceph/ceph.py

```diff
@@ -816,39 +820,39 @@ def setup_packages(
                     )
                 sleep(10)
 
-    def create_rbd_pool(self, k_and_m):
+    def create_rbd_pool(self, k_and_m, cluster_name=None):
         """
         Generate pools for later testing use
         Args:
             k_and_m(bool): ec-pool-k-m settings
         """
         ceph_mon = self.get_ceph_object("mon")
+
         if self.rhcs_version >= "3":
             if k_and_m:
                 pool_name = "rbd"
-                ceph_mon.exec_command(
-                    cmd="sudo ceph osd erasure-code-profile set %s k=%s m=%s"
-                    % ("ec_profile", k_and_m[0], k_and_m[2])
-                )
-                ceph_mon.exec_command(
-                    cmd="sudo ceph osd pool create %s 64 64 erasure ec_profile"
-                    % pool_name
-                )
-                ceph_mon.exec_command(
-                    cmd="sudo ceph osd pool set %s allow_ec_overwrites true"
-                    % (pool_name)
-                )
-                ceph_mon.exec_command(
-                    sudo=True,
-                    cmd="ceph osd pool application enable %s rbd --yes-i-really-mean-it"
-                    % pool_name,
-                )
+                commands = [
+                    f"ceph osd erasure-code-profile set ec_profile k={k_and_m[0]} m={k_and_m[2]}",
+                    f"ceph osd pool create {pool_name} 64 64 erasure ec_profile",
+                    f"ceph osd pool set {pool_name} allow_ec_overwrites true",
+                    f"ceph osd pool application enable {pool_name} rbd --yes-i-really-mean-it",
+                ]
+
+                if cluster_name is not None:
+                    commands = [
+                        f"{command} --cluster {cluster_name}" for command in commands
+                    ]
             else:
-                ceph_mon.exec_command(sudo=True, cmd="ceph osd pool create rbd 64 64 ")
-                ceph_mon.exec_command(
-                    sudo=True,
-                    cmd="ceph osd pool application enable rbd rbd --yes-i-really-mean-it",
-                )
+                commands = [
+                    "ceph osd pool create rbd 64 64",
+                    "ceph osd pool application enable rbd rbd --yes-i-really-mean-it",
+                ]
+            if cluster_name is not None:
+                commands = [
+                    f"{command} --cluster {cluster_name}" for command in commands
+                ]
```

Comment:
You would need to move this into the `else` block otherwise, we would see `--cluster {cluster_name}` appended twice when there is a value assigned to `k_and_m`. Alternatively, remove 841 - 844